src/shiver/views/workspace_tables.py

```diff
@@ -493,6 +520,67 @@ def set_pol_options(self, name):
         # at least one data workspace should be selected
         self.validate_data_workspace_state()
 
+    def clone_ws(self, name):
+        """method to clone the selected workspace"""
+
+        dialog = QInputDialog(self)
+        dialog.setLabelText(f"Clone {name} as:")
+        dialog.setTextValue(f"{name}_clone")
+        dialog.setOkButtonText("Clone")
+        if not dialog.exec_():
+            return
+
+        if self.clone_workspace_callback:
+            self.clone_workspace_callback(name, dialog.textValue())
+
+        # unselect the previous workspaces state of this workspace with name
+        self.unset_selected_states_with_name(name)
+
+        # at least one data workspace should be selected
+        self.validate_data_workspace_state()
+
+    def scale_ws(self, name):
+        """method to scale the workspace data, creating a new workspace"""
+
+        dialog = QDialog(self)
+        dialog.setWindowTitle("Scale MDE Workspace")
+
+        layout = QVBoxLayout()
+
+        # scale factor input
+        scale_factor_layout = QHBoxLayout()
+        scale_factor_layout.addWidget(QLabel("Scale Factor:"))
+        scale_factor_input = QLineEdit("1.0")
```

Comment:
scale_factor_input validation is missing. besides being a valid number is there another restriction on the number range? if yes, please add it.

Reply:
any news on this?

Reply:
CIS didn't request any input checking; said to let the user enter bad values

Reply:
It has to be a floating point number. The exact value does not matter